encodings/sparse/src/compute/search_sorted.rs

```diff
@@ -107,62 +118,31 @@ impl SearchSortedUsizeFn<&SparseArray> for SparseEncoding {
 
 #[cfg(test)]
 mod tests {
-    use rstest::{fixture, rstest};
-    use vortex_array::arrays::PrimitiveArray;
+    use rstest::rstest;
+    use vortex_array::compute::test_harness::rstest_reuse::apply;
+    use vortex_array::compute::test_harness::{search_sorted_conformance, *};
     use vortex_array::compute::{SearchResult, SearchSortedSide, search_sorted};
-    use vortex_array::validity::Validity;
     use vortex_array::{Array, ArrayRef, IntoArray};
     use vortex_buffer::buffer;
     use vortex_dtype::Nullability;
+    use vortex_error::VortexUnwrap;
     use vortex_scalar::Scalar;
 
     use crate::SparseArray;
 
-    fn sparse_high_null_fill() -> ArrayRef {
-        SparseArray::try_new(
-            buffer![17u64, 18, 19].into_array(),
-            PrimitiveArray::new(buffer![33_i32, 44, 55], Validity::AllValid).into_array(),
-            20,
-            Scalar::null_typed::<i32>(),
-        )
-        .unwrap()
-        .into_array()
-    }
-
-    fn sparse_high_non_null_fill() -> ArrayRef {
-        SparseArray::try_new(
-            buffer![17u64, 18, 19].into_array(),
-            buffer![33_i32, 44, 55].into_array(),
-            20,
-            Scalar::primitive(22, Nullability::NonNullable),
-        )
-        .unwrap()
-        .into_array()
-    }
-
-    fn sparse_low() -> ArrayRef {
-        SparseArray::try_new(
-            buffer![0u64, 1, 2].into_array(),
-            buffer![33_i32, 44, 55].into_array(),
-            20,
-            Scalar::primitive(60, Nullability::NonNullable),
-        )
-        .unwrap()
-        .into_array()
-    }
-
-    fn sparse_low_high() -> ArrayRef {
-        SparseArray::try_new(
-            buffer![0u64, 1, 17, 18, 19].into_array(),
-            buffer![11i32, 22, 33, 44, 55].into_array(),
-            20,
-            Scalar::primitive(30, Nullability::NonNullable),
-        )
-        .unwrap()
-        .into_array()
+    #[apply(search_sorted_conformance)]
+    fn sparse_search_sorted(
+        #[case] array: ArrayRef,
+        #[case] value: i32,
+        #[case] side: SearchSortedSide,
+        #[case] expected: SearchResult,
+    ) {
+        let sparse_array = SparseArray::encode(array).vortex_unwrap();
+        let res = search_sorted(&sparse_array, value, side).unwrap();
+        assert_eq!(res, expected);
     }
 
-    fn sparse_high_fill_in_patches() -> ArrayRef {
+    fn high_fill_in_patches() -> ArrayRef {
```

Comment:
the test cases that are left are unique to sparsearray, i.e. fill value exists in patches